crates/rover-client/src/query/schema/get.rs

```diff
@@ -0,0 +1,43 @@
+use crate::blocking::Client;
+use crate::RoverClientError;
+use graphql_client::*;
+
+// I'm not sure where this should live long-term
+/// this is because of the custom GraphQLDocument scalar in the schema
+type GraphQLDocument = String;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/query/schema/get.graphql",
+    schema_path = "schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. get_schema_query
+pub struct GetSchemaQuery;
+
+/// The main function to be used from this module. This function fetches a
+/// schema from apollo studio and returns it in either sdl (default) or json format
+pub fn run(
+    variables: get_schema_query::Variables,
+    client: Client,
+) -> Result<String, RoverClientError> {
+    let res = client.post::<GetSchemaQuery>(variables);
+
+    // if asking for a json response, try serializing the schema
+    // first unwrap the Result<Option<ResponseData>>
+    let data = res.expect("Error fetching schema");
+    let data = data.expect("No data in response when trying to fetch schema");
+
+    // get the schema document from ResponseData
+    let schema = data.service.expect("Service not found in response").schema;
+    let sdl = schema.expect("No schema found for this variant").document;
+
+    // if we want json, we can parse & serialize it here
+
+    Ok(sdl)
```

Comment:
I'm not sure what to do about error handling for data that's technically allowed to be optional. Like in the response for the get schema query, we need to access `data.service.schema.document` to get the sdl. But `data`, `service`, and `schema` can all technically be `None`.

Is there a good rusty way for accessing nested data like that other than what I do here? Is it possible to integrate these kinds of errors with `RoverClientError`?

Reply:
im not entirely sure what the original `data` type here looks like so i'd need to spend some time with that, but we likely dont want `expect` here as it will cause this result to panic if it doesn't exist (and it sounds like it's optional?) one thing you can do is call `.ok()` on your `Result` types which will return an `Option`, with `Some(T)` if the `Result` was `Ok` and `None` if it wasn't.

I'm not entirely sure if this solves your problem here since you'll still need to figure out what to do when it returns `None`.

Reply:
Yeah, it's a little hard to explain over text, but the CLI should _fail_ on any of those cases where anything along the way is `None`. I'm not quite sure how to best make it fail in a _nice_ way other than panicking like I did here. This is definitely what I want to pair on in the next couple days sometime

Reply:
I updated all the errors to be a little more explicit, and hopefully this prevents any of the panics :) I'd still like feedback on this, since it's a little verbose, and I'm not sure if there are any patterns to handle this kind of nested data structures well already